src/main/java/org/apache/commons/lang3/text/translate/NumericEntityUnescaper.java

```diff
@@ -77,12 +82,17 @@ public boolean isSet(final OPTION option) {
     /**
      * {@inheritDoc}
      */
+    @SuppressWarnings({"argument.type.incompatible","assignment.type.incompatible"}) /*
+    #1 if end == seqEnd, input.charAt(end) is not executed
+    #2 Character.toChars(x) where x > 0xFFFF returns an array of length at least 2
+    #3 False negative, entityValue does not need to be non negative
```

Comment:
Why is this a false negative? It looks like a false positive to me.

A false negative is a missed alarm: the typechecker should have issued an error, but did not. We take those very seriously, since the goal of the typechecker is to never miss errors.

By contrast, false positives (extra alarms about safe code) are common: the typechecker intentionally "forgets" information about the program in order to run in an acceptable amount of time. We want to have as few of these as possible, of course, but some *are* inevitable.